### DISCUSSION TITLE
How does concurreny work?

### DISCUSSION THREAD

**Discussion by b4ux1t3** [lichess-bot-devs/lichess-bot#794] [Q&A] [answered]
Does lichess-bot spin up a new engine for every new concurrent challenge, or am I expected to manage games within the bot itself? The docs just say that the number is  how many games can be played at once, there's no indication of whether that will be handled by discrete copies of the engine process, or if some other identifier is passed along with UCI commands that the engine has to juggle.

**Comment by mheinzel** [comment, marked as answer]
There is a separate engine process for each game.

**Reply by b4ux1t3** [reply]
That's awesome, I figured as much, but wanted to be sure before I bumped up the number.

Thanks a bunch for the quick answer!
